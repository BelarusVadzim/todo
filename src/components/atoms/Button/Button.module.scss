@import 'src/variables.scss';

.button {
  color: $secondary-font-color;
  cursor: pointer;
  font-family: $main-font;
  display: block;
  font-size: 16px;
  
&:hover {
  color: rgb(80, 80, 80);
  }
}
