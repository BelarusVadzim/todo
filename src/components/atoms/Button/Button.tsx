import React from "react";
import style from "./Button.module.scss";

type ButtonProps = {
  label: string;
  onClick: () => void;
};

const Button: React.FC<ButtonProps> = ({ label, onClick }) => (
  <span onClick={onClick} className={style.button}>
    {label}
  </span>
);

export default Button;
