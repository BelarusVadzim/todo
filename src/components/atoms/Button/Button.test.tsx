import render  from 'react-test-renderer';
import Button from './Button';

jest.mock('./Button.module.scss', () => ({ button: 'buttonStyle' }));

describe('<Button />', () => {
  it('should render properly', () => {
    const tree = render
      .create(<Button value='buttonValue' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when glyph className specified should render properly', () => {
    const tree = render
      .create(<Button glyph={true} value='buttonValue' className='className' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
