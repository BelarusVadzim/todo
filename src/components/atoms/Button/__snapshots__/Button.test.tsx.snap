// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly 1`] = `
<div
  className=" buttonStyle"
>
  buttonValue
</div>
`;

exports[`<Button /> when glyph className specified should render properly 1`] = `
<div
  className="material-icons buttonStyle className"
>
  buttonValue
</div>
`;
